core/src/main/java/org/itsallcode/openfasttrace/core/importer/ImporterFactoryLoader.java

```diff
@@ -63,18 +64,19 @@ public ImporterFactoryLoader(
      * @throws ImporterException
      *             when no or more than one {@link ImporterFactory} is found.
      */
-    public ImporterFactory getImporterFactory(final InputFile file)
+    public Optional<ImporterFactory> getImporterFactory(final InputFile file)
```

Comment:
Please split into check method and getter method instead of using `Optional`. Use getter only if check comes back green.

Reply:
This would mean that the ImporterFactory needs to be searched twice: once by the check method to figure out that there is  a fitting ImporterFactory and second by the get method. Th search calls ImporterFactory.supportsFile() on each factory when then figures out by examine the given file if it can load the file.

What is the problem with `Optional`. As Java has no `Either` construct, isn't Optional not the Java approach for such a construct?